### PR TITLE
Remove `*.mo` files from the repository.

### LINKED ISSUE #7: Remove the *.mo files from the repository
The *.mo files are now committed in the repository.
Unluckily there is no warranty that they are in sync with the *.po files.
This might lead to potential issues (see #5, #6).

### LINKED ISSUE #17: Release 4.2 no longer includes mo files
After #13 only [includes po/pot](https://github.com/zopefoundation/zope.app.locales/commit/deb4c8db3dee153ee3640bcf333ae2ba4fb12e5c#diff-4425ec4892a0640706a7612e3c26160478a52d1cb052428124b2e79769ef72d2R25-R26) in the release, causing the translations to not be available out-of-the-box anymore. Is this intentional? If yes, I'd appreciate a pointer on how consumers of this package are expected to use it now (compile to mo as part of our local application package release _somehow_?). Thanks!

### PR DESCRIPTION
**Pull request by icemac**
Fixes #7.

Document in the README how to generate `*.mo` files on the fly.

Fixes #17.

Create `*.mo` files during the release process.

This PR uses https://github.com/zopefoundation/meta/pull/154.

### CODE REVIEW

**Review by mauritsvanrees — CHANGES_REQUESTED** [review]
The mo files should not be in version control, so that part is good. The updated documentation is good.

But in my opinion the PyPI release *should* contain the compiled mo files. How? Install `zest.releaser` and `zest.pocompile` and run `fullrelease` Then the mo files get generated and included.
Well, `MANIFEST.in` would need `recursive-include src *.mo` otherwise they are not included after all.

See also https://github.com/zestsoftware/zest.pocompile

**Comment by icemac** [comment]
@mauritsvanrees If I understood correctly, `zest.pocompile` has to be installed next to `zest.releaser`. I am using a globally installed `zest.releaser`. I can install `zest.pocompile` there. But is there a way to ensure that `zest.pocompile` is installed once I re-install `zest.releaser` in another environment and have forgotten that `zest.pocompile` should also be there?

**Comment by mauritsvanrees** [comment]
I have not done this anywhere. But it is possible. cc @erral because this could be interesting for `plone.app.locales`: make sure running `fullrelease` fails when `zest.pocompile` is not installed. Let me check some options.

1. You could specify in `setup.cfg` that you want to use `zest.pocompile` and call it at the same time that it would normally run:

```
[zest.releaser]
releaser.after_checkout =
    zest.pocompile.compile.compile_in_tag
```

Then when you do not have it installed you get this error:

```
$ fullrelease
...
Traceback (most recent call last):
  File "/Users/maurits/zest.releaser/zest/releaser/utils.py", line 544, in resolve_name
    ret = getattr(ret, part)
AttributeError: module 'zest' has no attribute 'pocompile'
```

If you *do* have `zest.pocompile` installed, it will compile all po files that it finds. But after that, the *real* entry point is also triggered, and the po files are compiled *again*. So that is double work. Maybe `zest.releaser` should discover this corner case and only call it once.

And you only discover that you miss this package when you are halfway through the release. Preferably you would know it sooner.

2. Call it as the first hook:

```
[zest.releaser]
prereleaser.before =
    zest.pocompile.compile.compile_in_tag
```

This will give a warning:

    WARNING: Aborted compiling of po files: no tagdir found in data.

This is because this function is supposed to be called later, when the current working directory is in a clean temporary clone of the tag, which is what `zest.releaser` does right before uploading to PyPI.
The warning does no harm, but maybe you want to avoid it.

3. So call a different function in the beginning:

```
[zest.releaser]
prereleaser.before =
    zest.pocompile.compile.main
```

This will compile all po files that it finds.  That is actually a bit useless at this point though, because it does it on master. It should do it in the tag instead, as said. So this does not have an effect yet, except that it functions as a test that it works. And it does not take that long, so this could be a fine solution.

Alternatively, `zest.releaser` could suggest to authors of extensions to add a function `exists` or `require` or `available` that accepts a data dictionary as keyword arguments, and does completely nothing. You could use that then as a canary to check that the package is available. If that makes sense to you, I could do that in `zest.pocompile` now.

Otherwise, the third solution seems best to me.

**Comment by icemac** [comment]
@mauritsvanrees Thank you for your suggestion. I implemented the third item and tried without and with having `zest.pocompile` installed: It works, the release now again contains the `*.mo`. Could you please re-review?

File: README.rst [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
 messages can occur in Python code, in Page Templates and in ZCML
 declarations.  ``zope.app.locales`` provides a utility that can
 extract messages from all three and write them to a standard gettext
 template (``pot`` file).
+
+Usage
+=====
+
+Version 4.2 does not contain any ``*.mo`` files. They are the compiled
+translation files (``*.po``).
```
Comment (mauritsvanrees) [inline README.rst:16]: I would add: future versions will contain these files again.

**Review by mauritsvanrees — APPROVED** [review]

**Comment by mauritsvanrees** [comment]
LGTM now. I added one small comment about documentation, but this PR can be merged with or without it.

File: README.rst (hunk 1 above)
Comment (icemac) [reply README.rst]: Fixed in https://github.com/zopefoundation/zope.app.locales/pull/18/commits/d114d5465eb305bd21633a0e9b359ad0c8cf2f73.

**Pull request merged**